### PR TITLE
[registrar] Register models in the static registrar.

### PR DESCRIPTION
**Pull request by rolfbjarne**
This means we need to quiet a few types of warnings:.

* Models declares virtual methods of required protocol members. We don't
  export virtual methods (only when they're overridden are they exported),
  which results in numerous warnings about protocol members not being
  implemented:

        Xamarin.Mac.registrar.mobile.x86_64.m:37827:17: warning: method 'deviceBrowserView:selectionDidChange:' in protocol 'IKDeviceBrowserViewDelegate' not implemented [-Wprotocol]
        @implementation ImageKit_IKDeviceBrowserView__IKDeviceBrowserViewDelegate {
                        ^
        /Applications/Xcode92.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk/System/Library/Frameworks/Quartz.framework/Frameworks/ImageKit.framework/Headers/IKDeviceBrowserView.h:29:1: note: method 'deviceBrowserView:selectionDidChange:' declared here
        - (void)deviceBrowserView: (IKDeviceBrowserView *)deviceBrowserView selectionDidChange: (ICDevice *)device;

* These two are the same as above, just for properties instead of methods.

        Xamarin.Mac.registrar.mobile.x86_64.m:31988:17: warning: auto property synthesis will not synthesize property 'boundingMapRect' declared in protocol 'MKOverlay' [-Wobjc-protocol-property-synthesis]
        @implementation MKOverlay {
                        ^
        /Applications/Xcode92.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk/System/Library/Frameworks/MapKit.framework/Headers/MKOverlay.h:24:43: note: property declared here
        @property (nonatomic, readonly) MKMapRect boundingMapRect;
                                                  ^
        Xamarin.Mac.registrar.mobile.x86_64.m:32002:1: note: add a '@synthesize' directive
        @end
        ^

        Xamarin.Mac.registrar.mobile.i386.m:28957:17: warning: property 'repeatCount' requires method 'repeatCount' to be defined - use @synthesize, @dynamic or provide a method implementation in this class implementation [-Wobjc-property-implementation]
        @implementation CAMediaTiming {
                        ^
        /Applications/Xcode92.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk/System/Library/Frameworks/QuartzCore.framework/Headers/CAMediaTiming.h:58:17: note: property declared here
        @property float repeatCount;
                ^

It also means we have to special-case (to skip)
AVCaptureDataOutputSynchronizer[Delegate] in the registrar for macOS.

This class and protocol were incorrectly added to our macOS bindings, but
since we can't remove them because it would break backwards compatibility, we
must skip them manually in the registrar, since the registrar would otherwise
produce uncompilable code:

### CODE REVIEW

File: tools/common/StaticRegistrar.cs [hunk 1]
```diff
@@ -2529,8 +2529,13 @@ void Specialize (AutoIndentStringBuilder sb)
...
 					continue;
 
-				if (@class.IsModel)
-					continue;
+#if MMP
+				switch (@class.ExportedName) {
+				case "AVCaptureDataOutputSynchronizer":
+				case "AVCaptureDataOutputSynchronizerDelegate":
+					continue; // We've accidentally added bindings for these two classes which are not available in macOS.
+				}
```
Comment (spouliot) [inline tools/common/StaticRegistrar.cs]: Can we just stub them on macOS ? i.e. they don't need the attributes or export/selectors that would make the static registrar triggers on them ?

File: tools/common/StaticRegistrar.cs [hunk 2]
```diff
@@ -2717,6 +2730,8 @@ void Specialize (AutoIndentStringBuilder sb)
 					}
 					sb.Unindent ();
 					sb.WriteLine ("@end");
+					if (hasClangDiagnostic)
+						sb.AppendLine ("#pragma clang diagnostic pop");
```
Comment (spouliot) [inline tools/common/StaticRegistrar.cs]: 👍

**Comment by rolfbjarne** [comment]
build

**Comment by rolfbjarne** [comment]
Test failures are unrelated (xtro: https://github.com/xamarin/maccore/issues/627)

**Pull request merged**